src/main/java/sirius/biz/model/QueryController.java

```diff
@@ -186,8 +190,11 @@ private List<Property> determineVisibleProperties(EntityDescriptor descriptor) {
     @Routed("/system/query/api")
     @InternalService
     public void queryApi(WebContext webContext, JSONStructuredOutput output) {
-        if (!Sirius.isDev() && !Sirius.isTest()) {
-            throw new IllegalStateException("The query API is only available in development or test systems.");
+        if (!queryApiEnabled) {
+            throw new IllegalStateException("The query API is not enabled in this system.");
+        }
+        if (Sirius.isProd()) {
+            throw new IllegalStateException("The query API is not available in productive systems.");
```

Comment:
```suggestion
        if (!queryApiEnabled) {
            throw new IllegalStateException("The query API is not enabled on this system.");
        }
        if (Sirius.isProd()) {
            throw new IllegalStateException("The query API is not available on productive systems.");
```
Sounds better (and maybe is only correct?) this way.

Reply:
The wise Internet says:
_Both “in the system” and “on the system” are correct. However, they have different meanings. “In the system” typically means that something is part of or related to the system. On the other hand, “on the system” typically means that something is being done with or using the system._

Reply:
Fascinating, so basically, "in the system" is even more correct here...